Wendy/assignment1Answers.html

```diff
@@ -0,0 +1,219 @@
+
+					<h3>f) x = true  true.</h3>
+					<p>  Assigning the boolean value of true to x.</p>
+				</div>
+
+				<div class="col-md-4">
+					<h3>g) var x; x == 'undefined'  false.</h3> 
+					<p>It seems like the intent was to initialize a variable but leave it as undefined.  However, variables are already undefined until they are initialzed.  Moreover, the =='undefined' is shown as a string not a value.  Undefined would be a value not a string.</p>  
+
+					<h3>h) '9'<'10' false.</h3>
+					<p>  '' makes these strings not numbers and so they inherently do not have values.</p>
+
+					<h3>i) typeof x + 1 === "number" false. </h3> <p>typeof returns what type of variable x is.  If x is a number, the return would be "number1"-- it concatenates.  If x were a string, the return would be "string1".  Typeof is going to return a description not literally what it is.</p>
+
+					<h3>j) typeof x % 2 === "number" nope.</h3> 
+					<p>Still false.  This isn't the right way to use the typeof operator.  It is used to determine the type of JavaScript variable.</p>
+
+					<h3>k) typeof (x % 2) === "number"  true.</h3><p>Javascript reads from left to right but the paranthesis group x with modulo 2.  Seems that modulo IS a number so even if x were defined as a string and the second value was also a string instead of the number 2, it would still be correct to have the === be "number".</p>
+
+					<h3>l) x++ == ++x false. </h3> 
+					<p>x++ Is typically seen in loops and means to increment.  ++x would mean addition operator and positive number x.  So x++ is typically increment by one, plus positive x would not necessarily be the same thing.</p>
+	
+				</div>
+				<div class="col-md-4">
+					<h3>m) ++x == x++ false if x is a string.  true if x is a number.</h3>
+					<p>  True because ++x is saying "plus positive x" and that is sort of the same as x++ (incrementing).</p>
+
+					<h3>n) "1"+x == 1+x  false. </h3>
+					<p> Javascript sees that 1 is a string and so it's going to want to concatenate.  The answer on the right should thus be 1 concatenated with whatever x was. . . for example, if x = 6, "1" + x would equal "16".  The answer on the right is shown as numbers.</p>
+
+					<h3>o) "0"+1 == 1  true.  </h3>
+					<p>Javascript can ignore entirely the 0 in this case, it's a string.  And since we're asking if this is kind of true with just the == and not the === then it is kind of true.  JS is assuming that the string "0" has no value.</p>
+
+					<h3>p) (typeof (x+1))===(typeof x) true. </h3><p>This is just comparing the types of variables.  Whethers numbers or strings, it's still asking if x is the same type as x.</p>
+
+					<h3>q) (x*1 == x) || ((typeof x) != "number") true.</h3>
+					<p>  the || means if EITHER of these statements is true, than the whole statement is true.  If x is a number, both statemetns are true.  If x is a string, then the first statement is false because (x*1) is undefined.</p>
```

Comment:
You're very close on this one. If x is a number, however, the right side of the equation will not actually be evaluated at all. As soon as it goes through (x*1 ==x) its requirements are met and it simply returns true. On the off chance that it did not meet its requirements as a number on the left side, it would ultimately return false. The reason is that (again assuming x is a number) typeof x would return "number" which === "number, not satisfying the condition of (typeof x) != "number".